src/content/en/fundamentals/building-components/best-practices.md

```diff
@@ -0,0 +1,535 @@
+        </h3>
+      </th>
+    </tr>
+    <tr>
+      <td><b>Why?</b></td>
+      <td>
+  You never know how a user will try to interact with your element. They might
+  set a property in JavaScript, and then expect to be able to read that value
+  using an API like <code>getAttribute</code>. If every attribute has a
+  corresponding property, and both of them reflect, it will make it easier for
+  users to work with your element. In other words, calling
+  <code>setAttribute('foo', value)</code> should also set a corresponding
+  <code>foo</code> property. And vice versa. There are, of course, exceptions to
+  this rule. You shouldn't reflect high frequency properties, e.g.
+  <code>currentTime</code> in a video player. Use your best judgment. If it
+  seems like a likely a user will interact with a property or attribute, and
+  it's not burdensome to reflect it, then do so.
+      </td>
+    </tr>
+    <tr>
+      <td><b>Example</b></td>
+      <td>
+  The <a href="/web/fundamentals/building-components/examples/howto-checkbox">
+  <code>&lt;howto-checkbox&gt;</code></a> element. Further explained in
+  <a href="#avoid-reentrancy">Avoid reentrancy issues</a>.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+<table class="responsive ce-cl">
+  <tbody>
+    <tr>
+      <th colspan=2>
+        <h3>
+  Aim to only accept rich data (objects, arrays) as properties.
+        </h3>
+      </th>
+    </tr>
+    <tr>
+      <td><b>Why?</b></td>
+      <td>
+  Generally speaking, there are no example of built-in HTML Elements that accept
+  rich data (plain JavaScript objects and arrays) through their attributes. Rich
+  data is instead accepted either through method calls or properties. There are
+  a couple obvious downsides to accepting rich data as attributes: it can be
+  expensive to serialize a large object to a string, and any object references
+  will be lost in this stringification process. For example, if you stringify an
+  object which has a reference to another object, or perhaps a DOM node, those
+  references will be lost.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+<table class="responsive ce-cl">
+  <tbody>
+    <tr>
+      <th colspan=2>
+        <h3>
+  Do not reflect rich data properties to attributes.
+        </h3>
+      </th>
+    </tr>
+    <tr>
+      <td><b>Why?</b></td>
+      <td>
+  Reflecting rich data properties to attributes is needlessly expensive,
+  requiring serializing and deserializing the same JavaScript objects. Unless
+  you have a specific use case that this feature solves, it's probably best to
+  avoid it.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+<table class="responsive ce-cl">
+  <tbody>
+    <tr>
+      <th colspan=2>
+        <h3>
+  Consider checking for properties that may have been set before the element
+  upgraded.
+        </h3>
+      </th>
+    </tr>
+    <tr>
+      <td><b>Why?</b></td>
+      <td>
+  A developer may attempt to set a property on the element before its
```

Comment:
Changed this to:

"A developer using your component may"

I'm trying to refer to when you build a component and then give it to somebody else on your team who uses it in a way you maybe didn't expect.